usaspending_api/search/v2/views/spending_by_category_views/base_spending_by_category.py

```diff
@@ -0,0 +1,153 @@
+import copy
+import itertools
+import logging
+from abc import ABCMeta, abstractmethod
+from dataclasses import dataclass
+from typing import List
+
+from django.conf import settings
+from django.db.models import QuerySet, Sum
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from usaspending_api.awards.v2.filters.sub_award import subaward_filter
+from usaspending_api.awards.v2.filters.view_selector import spending_by_category_view_queryset
+from usaspending_api.common.api_versioning import api_transformations, API_TRANSFORM_FUNCTIONS
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.elasticsearch.search_wrappers import TransactionSearch
+from usaspending_api.common.experimental_api_flags import is_experimental_elasticsearch_api
+from usaspending_api.common.helpers.generic_helper import get_simple_pagination_metadata, get_time_period_message
+from usaspending_api.common.query_with_filters import QueryWithFilters
+from usaspending_api.common.validator.award_filter import AWARD_FILTER
+from usaspending_api.common.validator.pagination import PAGINATION
+from usaspending_api.common.validator.tinyshield import TinyShield
+from usaspending_api.search.v2.elasticsearch_helper import get_number_of_unique_terms
+
+logger = logging.getLogger("console")
+
+
+@dataclass
+class Category:
+    name: str
+    primary_field: str
+
+
+@api_transformations(api_version=settings.API_VERSION, function_list=API_TRANSFORM_FUNCTIONS)
+class BaseSpendingByCategoryViewSet(APIView, metaclass=ABCMeta):
+    """
+    Base class inherited by the different category endpoints.
+    """
+
+    category: Category
+    elasticsearch: bool
+    filters: dict
+    obligation_column: int
+    pagination: dict
+    subawards: bool
+
+    @cache_response()
+    def post(self, request: Request) -> Response:
+        models = [
+            {"name": "subawards", "key": "subawards", "type": "boolean", "default": False, "optional": True},
+        ]
+        models.extend(copy.deepcopy(AWARD_FILTER))
+        models.extend(copy.deepcopy(PAGINATION))
+
+        validated_payload = TinyShield(models).block(request.data)
+
+        validated_payload["elasticsearch"] = is_experimental_elasticsearch_api(request)
+
+        return Response(self.handle_business_logic(validated_payload))
+
+    def handle_business_logic(self, validated_payload: dict) -> dict:
```

Comment:
```suggestion
    def perform_search(self, validated_payload: dict) -> dict:
```
happy to back off on this one since "handle_business_logic" is the norm among our endpoints, but I think there's an opportunity for self-documenting code here.